app/Rules/ReCaptchaValidation.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace App\Rules;
+
+use Illuminate\Contracts\Validation\ImplicitRule;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+use \ReCaptcha\ReCaptcha;
+
+class ReCaptchaValidation implements ImplicitRule
+{
+    /**
+     * @var \ReCaptcha\ReCaptcha $recaptcha  instance
+     */
+    protected $recaptcha;
+
+    /**
+     * @var string $minScore Lowest score to pass (0.0 - 1.0)
+     */
+    protected $minScore;
+
+    /**
+     * @var string $appUrl App URL of client request
+     */
+    protected $appUrl;
+
+    public function __construct(ReCaptcha $recaptcha, $minScore, $appUrl) {
+        $this->recaptcha = $recaptcha;
+        $this->minScore = $minScore;
+        $this->appUrl = $appUrl;
+    }
+
+    /**
+     * Comparison against expected hostname.
+     * 
+     * @param  string $hostname
+     * @return bool  
+     */
+    private function verifyHostname($hostname)
```

Comment:
I just realised here: for example you could have type hinted using more than just docstring (e.g. actually type bot the parameters and return types; no need to fix but maybe we want to start linting for this for consistency